File: USACO/Guide/Sets_and_Maps/whereami.py
with open("whereami.in") as file:
    raw = file.read().split("\n")
    n = int(raw[0])
    mailboxes = raw[1]
result = 0
for i in range(1,n+1):
    seqs = []
    unique = True
    for j in range(n-i+1):
        current = mailboxes[j:j+i]
        if current in seqs:
            unique = False
            break
        seqs.append(current)
    if unique:
        result = i
        break
print(result)
with open("whereami.out", "w") as out:
    out.write(str(result))

File: USACO/Guide/Sets_and_Maps/whereami.java
import java.util.*;
import java.io.*;

public class whereami {
	public static void main(String[] args) throws FileNotFoundException {
		Scanner in = new Scanner(new File("whereami.in"));
		int n = in.nextInt();
		String mailboxes = in.next();
		ArrayList<String> seqs = new ArrayList<>();
		for(int i = 1; i<=n; i++){
			boolean unique = true;
			for(int j = 0; j<=n-i; j++){
				String current = mailboxes.substring(j,j+i);
				if(seqs.contains(current)){
					unique = false;
					break;
				}
				seqs.add(current);
			}
			if(unique){
				System.out.println(i);
				PrintWriter out = new PrintWriter(new File("whereami.out"));
				out.println(i);
				out.close();
				in.close();
				System.exit(0);
			}
		}
	}
}